apis/Google.Cloud.PubSub.V1/docs/index.md

```diff
@@ -82,6 +82,99 @@ restart listening for messages with `StartAsync(...)` again. Due to the expense
+using the [`AddHostedService`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.servicecollectionhostedserviceextensions.addhostedservice?view=dotnet-plat-ext-6.0) extension method as shown below:
+
+{{sample:SubscriberClient.AddHostedService}}
+
+Here `SubscriberService` is the class that implements `BackgroundService` and uses the `SubscriberClient`
+registered with the dependency injection container to handle the messages. Once the background service is registered,
+it will be automatically started when the application starts and stopped when the application exits.
+A sample implementation of `SubscriberService` is shown below:
+
+{{sample:SubscriberClient.UseSubscriberClient}}
+
+During application shutdown, the `StopAsync` method of the `SubscriberService` is invoked by the dependency injection container, which in turn calls
+the `StopAsync` method of the `SubscriberClient` to gracefully shut down the client.
+
+Below is an example implementation of a console application that utilizes the dependency injection container and the `SubscriberService` to handle messages:
+
+{{sample:SubscriberClient.UseSubscriberServiceInConsoleApp}}
+
+## Disposing of the publisher and subscriber clients
+
+Both `PublisherClient` and `SubscriberClient` implement the `IAsyncDisposable` interface,
+allowing for asynchronous resource cleanup.
+
+For the `PublisherClient`, when the `DisposeAsync` method is called, it asynchronously waits for the time interval
+specified in the `PublisherClient.Settings.DisposeTimeout` property for the `PublisherClient` to send any pending messages before aborting
+the clean shutdown process, that may leave some locally queued messages unsent. The time interval can be customized
+by setting the `PublisherClient.Settings.DisposeTimeout` property. If no value is specified, the default value of 5 seconds is used.
+
+For the `SubscriberClient`, when the `DisposeAsync` method is called, it asynchronously waits for the time interval
+specified in the `SubscriberClient.Settings.DisposeTimeout` property for the `SubscriberClient` to acknowledge the handled messages before aborting the
+the clean stop process, that may leave some handled messages un-acknowledged. The time interval can be customized
+by setting the `SubscriberClient.Settings.DisposeTimeout` property. If no value is specified, the default value of 5 seconds is used.
+
+Please note that when working with the dependency injection container, if the timeout interval specified in the properties above is greater than the default value,
+the dependency injection container must be configured to have a timeout greater than or equal to this time. Failure to do so may result in the abrupt termination of the client shutdown process.
+
+See [ShutdownTimeout](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/host/generic-host?view=aspnetcore-6.0#shutdowntimeout) property for configuring the shutdown timeout in ASP.NET Core applications.
```

Comment:
```suggestion
See [HostOptions.ShutdownTimeout](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/host/generic-host?view=aspnetcore-6.0#shutdowntimeout) for configuring the shutdown timeout in ASP.NET Core applications.
```

(Or have "the HostOptions.ShutdownTimeout property" if you want to include the word "property".